src/evict.c

```diff
@@ -412,6 +412,21 @@ int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *lev
     return C_ERR;
 }
 
+/* This function is to check whether used memory is more than maxmemory after
+ * allocating more memory. Redis may reject user's requests or evict some keys
+ * if used memory exceeds maxmemory, especially, when we allocate huge memory
+ * at same time. */
+int overMaxmemoryAfterAlloc(size_t moremem) {
+    if (!server.maxmemory) return 1; /* No limit. */
```

Comment:
i think all the return values of this function should be negated, or the function renamed.
i.e. to me the name indicates that it returns true when the new allocation will be over the limit.